### PR TITLE
Upload live test coverage to codecov

### PR DESCRIPTION
**Pull request by jayqi**
Right now only mocked tests are reflected in codecov.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -113,6 +113,12 @@ jobs:
           CUSTOM_S3_SECRET_KEY: ${{secrets.CUSTOM_S3_SECRET_KEY}}
           CUSTOM_S3_ENDPOINT: ${{secrets.CUSTOM_S3_ENDPOINT}}
 
+      - name: Upload coverage to codecov
+        uses: codecov/codecov-action@v1
+        with:
+          file: ./coverage.xml
+          fail_ci_if_error: true
```
Comment (pjbull) [inline .github/workflows/tests.yml:120]: Does codecov merge these for us rather than having two different reports?

Comment (pjbull) [reply .github/workflows/tests.yml:120]: I guess if coverage goes up above then we've got a merge scenario

Comment (jayqi) [reply .github/workflows/tests.yml:120]: Yes, codecov merges them.

Comment (jayqi) [reply .github/workflows/tests.yml:120]: https://docs.codecov.com/docs/merging-reports

**Pull request merged**